spec/xref_spec.rb

```diff
@@ -37,12 +37,11 @@
       include::chapter-b.adoc[]
       EOS
 
-      doc = Asciidoctor.load_file book_file, backend: 'epub3', header_footer: true
-      doc.convert
+      doc = Asciidoctor.convert_file book_file, backend: 'epub3', header_footer: true
       spine_items = doc.references[:spine_items]
       (expect spine_items).to have_size 2
       chapter_a_content = doc.references[:spine_items][0].content
-      (expect chapter_a_content).to include '<a id="xref--chapter-b" href="chapter-b.xhtml" class="xref">Chapter B</a>'
+      (expect chapter_a_content).to include '<a href="chapter-b.xhtml" class="xref">Chapter B</a>'
```

Comment:
It shouldn't be missing the id attribute.

Reply:
But it does :( Will investigate why that is happening (and what happens when we use asciidoctor-epub3 through CLI).

Reply:
Real book, created by `asciidoctor-epub3` CLI, does have `id="xref-bla-bla` attributes. Investigating why that doesn't happen with changes from this PR...

Reply:
1. We have [logic that strips away link id](https://github.com/asciidoctor/asciidoctor-epub3/blob/master/lib/asciidoctor-epub3/converter.rb#L687) if converter detects mulltiple attempts to convert links with identical id.
2. When test [gets chapter content](https://github.com/asciidoctor/asciidoctor-epub3/blob/master/spec/xref_spec.rb#L44) it is actually triggering conversion using the same converter that was used to convert document initially! So, test gets a _different_ output from what actually went into epub.

I'm not sure yet what to do with this.

Reply:
I'll leave test as-is, but will add clarifying comments.

Reply:
See ba969f8ac0461445d3cf5658da62579c60609e8a.

Reply:
It's probably an indication that either a) the converter needs to be resilient against side effects from this call order or b) the test harness needs to be improved to abstract this call order away so that the test doesn't have to think about the right way to do it.